src/cgame/etj_jump_speeds.cpp

```diff
@@ -0,0 +1,202 @@
+		DrawString(x1, y1, 0.2f, 0.2f, color, qfalse, "Jump Speeds:", 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+
+		// adjust x or y depending on style chosen
+		vertical ? y1 += 12 : x1 = DrawStringWidth("Jump Speeds: ", 0.2f) + 5;
+
+		for (auto i = 0; i < jumpSpeedHistory.size(); i++)
+		{
+			auto jumpSpeed = std::to_string(jumpSpeedHistory.at(i));
+			if (etj_jumpSpeedsShowDiff.integer)
+			{
+				AdjustColors(i, &color);
+			}
+			if (vertical)
+			{
+				// first column
+				if (i < 5)
+				{
+					DrawString(x1, y1, 0.2f, 0.2f, color, qfalse, jumpSpeed.c_str(), 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+					y1 += 12;
+				}
+				// second column
+				else
+				{
+					DrawString(x2, y2, 0.2f, 0.2f, color, qfalse, jumpSpeed.c_str(), 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+					y2 += 12;
+				}
+			}
+			// horizontal list
+			else
+			{
+				DrawString(x1, y1, 0.2f, 0.2f, color, qfalse, jumpSpeed.c_str(), 0, drawShadow ? ITEM_TEXTSTYLE_SHADOWED : ITEM_TEXTSTYLE_NORMAL);
+				x1 += 30;
+			}
+		}
+	}
+
+	void UpdateJumpSpeeds()
+	{
+		playerState_t* ps = getValidPlayerState();
+		// queue reset if last update was on different team
+		if (team != ps->persistant[PERS_TEAM])
+		{
+			resetQueued = true;
+		}
+		// if reset is queued, do that before we start storing new jump speeds
+		if (resetQueued)
+		{
+			ResetJumpSpeeds();
+			resetQueued = false;
+		}
+
+		team = ps->persistant[PERS_TEAM];
+		jumpSpeedHistory.push_back(ps->persistant[PERS_JUMP_SPEED]);
+		// we only want to keep last 10 jumps, so remove first value if we go over that
+		if (jumpSpeedHistory.size() > MAX_JUMPS)
+		{
+			// store the deleted speed so we can do diff comparisons later
+			lastDeletedSpeed = jumpSpeedHistory.front();
+			jumpSpeedHistory.erase(jumpSpeedHistory.begin());
+			jumpSpeedDeleted = true;
+		}
+	}
+
+	void QueueJumpSpeedsReset()
+	{
+		resetQueued = true;
+	}
+
+	void ResetJumpSpeeds()
+	{
+		jumpSpeedHistory.clear();
+		lastDeletedSpeed = 0;
+	}
+
+	void AdjustColors(int jumpNum, vec4_t* color)
+	{
+		auto alpha = (*color)[3];
+		vec4_t fasterColor = { 0.0, 1.0, 0.0, alpha };
+		vec4_t slowerColor = { 1.0, 0.0, 0.0, alpha };
+		auto currentJumpSpeed = jumpSpeedHistory[jumpNum];
+
+		// when checking for the first jump speed, we need to make sure
+		// we don't do comparison of (first jump - 1). Instead, compare against
+		// previously deleted jump speed, but only if the history is full
+		if (jumpNum == 0)
+		{
+			if (jumpSpeedHistory.size() == MAX_JUMPS && jumpSpeedDeleted)
+			{
+				// faster than previous jump
+				if (currentJumpSpeed > lastDeletedSpeed)
+				{
+					Vector4Copy(fasterColor, *color);
+				}
+				// slower than previous jump
+				else if (currentJumpSpeed < lastDeletedSpeed)
+				{
+					Vector4Copy(slowerColor, *color);
+				}
+				// equal speed
+				else
+				{
+					Vector4Set(*color, 1.0, 1.0, 1.0, alpha);
```

Comment:
Doesn't seem like this function respects `etj_jumpSpeedsColor`?

Reply:
This is intended, when `etj_jumpSpeedsShowDiff` is enabled, jumps are colored either white (first jump/equal speed), green (faster), or red (slower).